lib/src/calendar.dart

```diff
@@ -547,7 +548,10 @@ class _TableCalendarState extends State<TableCalendar> with SingleTickerProvider
   }
 
   TableRow _buildTableRow(List<DateTime> days) {
-    return TableRow(children: days.map((date) => _buildTableCell(date)).toList());
+    return TableRow(
+      decoration: widget.calendarStyle.dayRowBoxDecoration,
```

Comment:
This one I don't like. It will be better to wrap the `Table` with `DecoratedBox` instead. Also, please rename as per earlier comment.

Reply:
Both this and the decoration above are for rows which are part of the same table so I'm not sure what you mean

Reply:
Correct, but here you create a separate decoration for each row (copying the same decor many times), while having a single decoration in parent widget will apply it just once.

Difference becomes obvious when you add a `border` with `borderRadius`. This would wrap each row with a separate border; parent widget would wrap the whole content as a single entity.

Reply:
The intention here is to wrap individuals rows only, not the whole table. You do give the whole table a background by setting a background on each row but the idea is it could also be a border on each row, or whatever else.

Reply:
it might make more sense to add a callback / builder that lets them have an index so they can select which row(s) they want to apply they changes to but that is outside the scope of my PR and can be done separately

Reply:
Let's go with the current approach